pkg/lib/cockpit-components-listing.jsx

```diff
@@ -187,11 +192,18 @@ export class ListingRow extends React.Component {
                 clickHandler = this.handleExpandClick;
         }
 
+        let checkboxItem;
+        if (allowSelect && this.props.checkboxVisible == true) {
+            checkboxItem = <td key="checkboxItem" className="listing-ct-toggle" >
+                <input type='checkbox' checked={this.state.selected || false} onChange={this.handleSelectClick} />
+            </td>;
+        }
+
         let listingItem = (
             <tr data-row-id={ this.props.rowId }
                 className={ listingItemClasses.join(' ') }
                 onClick={clickHandler}>
-                {expandToggle}
+                {checkboxItem || expandToggle}
```

Comment:
The reason I had this like this is that ```Listing``` component always has only **one** extra header aside with title columns. This can be seen here. https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/cockpit-components-listing.jsx#L308

So I wanted to make sure to have only one extra column as well, so that we 'll not break the arithmetic of the columns.
But it's not clean solution indeed, I 'll repost with changes.